src/Blocks/BlockPlant.h

```diff
@@ -1,32 +1,41 @@
 
-// BlockPlant.h
-
-// Base class for any growing block
-
+#pragma once
 
+#include "BlockHandler.h"
 
 
 
-#pragma once
 
-#include "BlockHandler.h"
 
-class cBlockPlant : public cBlockHandler
+/** Base class for plants that use light values to decide whether to grow or not. */
+template <bool NeedsLightToGrow>
+class cBlockPlant:
```

Comment:
Is there any reason for this to be a template rather than a derived class that overrides `CanGrow`?

Reply:
Only me not wanting to refactor more than needed.

This should probably be refactored into a base class that reads the light and sends it to a virtual `checkLight` function that would be overridden in two descendants, `cBlockPlantWantsLight` and `cBlockPlantHatesLight` that override just the `checkLight` function.

Reply:
If you derive `cBlockPlantWantsLight` from `cBlockPlantHatesLight` (or vice versa) then there should be no duplicated code.

Reply:
Deriving one from another feels wrong, basic OOP says you don't derive a dog from a cat, you derive both from an animal. Still, outside the scope for this PR, I think. I just needed to get rid of the bool param to the constructor, so that I can use it as a mixin.